File: Data Structures & Algorithms/binary-tree-diameter/submission-0.cs
/**
 * Definition for a binary tree node.
 * public class TreeNode {
 *     public int val;
 *     public TreeNode left;
 *     public TreeNode right;
 *     public TreeNode(int val=0, TreeNode left=null, TreeNode right=null) {
 *         this.val = val;
 *         this.left = left;
 *         this.right = right;
 *     }
 * }
 */

public class Solution {
    int res = 0;
    public int DiameterOfBinaryTree(TreeNode root) {
        if (root == null) return 0;
        DFS(root);

        return res;
    }

    public int DFS(TreeNode node)
    {
        if (node == null) return 0;

        int leftHeight = DFS(node.left);
        int rightHeight = DFS(node.right);

        res = Math.Max(res, leftHeight + rightHeight);

        return 1 + Math.Max(leftHeight, rightHeight);
    }
}

File: Data Structures & Algorithms/longest-repeating-substring-with-replacement/submission-0.py
class Solution:
    def characterReplacement(self, s: str, k: int) -> int:
        d = defaultdict(int)
        maxF = 0
        l=0
        res = 0
        for r in range(len(s)):
            d[s[r]]+=1
            maxF = max(maxF, d[s[r]])

            while (r-l+1) - maxF > k:
                d[s[l]]-=1
                l+=1
            
            res = max(res, r-l+1)
        
        return res
            


                

                

